sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java

```diff
@@ -477,7 +476,7 @@ private void startStatsbeatModule(StatsbeatModule statsbeatModule, ConfigPropert
             initStatsbeatFeatures());
     }
 
-    private TelemetryItemExporter createTelemetryItemExporter(StatsbeatModule statsbeatModule) {
+    private static TelemetryItemExporter createTelemetryItemExporter(HttpPipeline httpPipeline, StatsbeatModule statsbeatModule) {
         TelemetryPipeline telemetryPipeline = new TelemetryPipeline(httpPipeline, statsbeatModule);
```

Comment:
the problem was the `httpPipeline` here on this (unchanged) line used to reference the (user populated) field instead of the `HttpPipeline` that was just built

Reply:
too bad our test didn't catch it.

Reply:
let's discuss, it may be worth pulling over our fake ingestion endpoint from https://github.com/microsoft/ApplicationInsights-Java/tree/main/smoke-tests/framework so we can have real end-to-end tests